From e9f89556ebbc1e281d7bdb51da7d3c8e13c6c379 Mon Sep 17 00:00:00 2001
From: TomZ <tom@flowee.org>
Date: Fri, 26 Jul 2024 11:49:46 +0200
Subject: [PATCH] Backport uPNP compile fix

This comes from Bitcoin Core commit: 8acdf6654

upnp: add compatibility for miniupnpc 2.2.8
original patch by: Cory Fields
---
 hub/server/net.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/hub/server/net.cpp b/hub/server/net.cpp
index c524be2a9..bed48b483 100644
--- a/hub/server/net.cpp
+++ b/hub/server/net.cpp
@@ -1356,7 +1356,11 @@ void ThreadMapPort()
     struct IGDdatas data;
     int r;
 
+#if MINIUPNPC_API_VERSION <= 17
     r = UPNP_GetValidIGD(devlist, &urls, &data, lanaddr, sizeof(lanaddr));
+#else
+    r = UPNP_GetValidIGD(devlist, &urls, &data, lanaddr, sizeof(lanaddr), nullptr, 0);
+#endif
     if (r == 1)
     {
         if (fDiscover) {
-- 
2.45.2

